base_tier_validation/models/tier_review.py

```diff
@@ -20,6 +20,14 @@ class TierReview(models.Model):
     )
     model = fields.Char(string="Related Document Model", index=True)
     res_id = fields.Integer(string="Related Document ID", index=True)
+    record_id = fields.Reference(
+        selection=lambda self: [
+            (m.model, m.name) for m in self.env["ir.model"].search([])
```

Comment:
maybe a search_read on model and name only is better?

Reply:
I don't totally agree with the `search_read` here since it would return a list of dictionaries including the ids of every record which it seems a little too much for me